src/theory/quantifiers/sygus/synth_conjecture.cpp

```diff
@@ -539,7 +539,8 @@ bool SynthConjecture::doCheck(std::vector<Node>& lems)
     return false;
   }
 
-  lem = Rewriter::rewrite(lem);
+  // simplify the lemma based on the term database sygus utility
+  lem = d_tds->rewriteNode(lem);
```

Comment:
Shouldn't you already be replacing other instances of lemma rewriting within the sygus module? There are several others that you have not rewritten.

Reply:
That is a good point, let's do this in a separate PR once we resolve the new default behavior of the sygus solver. I think many places will change (everywhere that calls the extended rewriter).

Reply:
OK